backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -149,7 +164,134 @@ export class SegmentService {
     return await this.segmentRepository.deleteSegment(id, logger);
   }
 
-  public async importSegments(segments: SegmentInputValidator[], logger: UpgradeLogger): Promise<Segment[]> {
+  public async importSegments(segments: SegmentFile[], logger: UpgradeLogger): Promise<SegmentReturnObj> {
+    const importFileErrors: SegmentImportError[] = [];
+    if (this.isJSONString(segments[0].fileContent)) {
```

Comment:
You are checking only the first segment and that also without the confirmation that the `segments` length is greater than zero

Reply:
Can you change the code flow to this
```
for each segment
 if type json
  parseJson
 else
  parseCsv
importSegment
```